apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/repository/datasource/remote/RemoteStorage.kt

```diff
@@ -15,7 +16,31 @@ class RemoteStorage @Inject constructor(private var liveVideoAPIService: LiveVid
             result.code = it.code()
             result.isApiResponseSuccess = it.errorBody() == null
             if (!it.isSuccessful) { // Failure case
-                ApiResponseUtil.parseErrorBody(gson, it.errorBody(), result, VerifyPasscodeResponse::class.java)
+                ApiResponseUtil.parseErrorBody(
+                    gson,
+                    it.errorBody(),
+                    result,
+                    VerifyPasscodeResponse::class.java
+                )
+            }
+            result
+        }
+
+    suspend fun joinStreamAsViewer(
+        userIdentity: String,
+        streamName: String
+    ): JoinStreamAsViewerResponse =
+        liveVideoAPIService.joinStreamAsViewer(userIdentity, streamName).let {
+            val result: JoinStreamAsViewerResponse = it.body() ?: JoinStreamAsViewerResponse("")
+            result.code = it.code()
+            result.isApiResponseSuccess = it.errorBody() == null
+            if (!it.isSuccessful) { // Failure case
+                ApiResponseUtil.parseErrorBody(
+                    gson,
+                    it.errorBody(),
+                    result,
+                    JoinStreamAsViewerResponse::class.java
+                )
             }
```

Comment:
I think this is duplicate code that each request has. 

You don't have to refactor it in this PR. But I would start to think about ways to refactor this in an upcoming PR when you add more network requests.

Hopefully there is a way to make an API client that reuses more code. I have some ideas and could show you what iOS does if you want to chat sometime.

Reply:
Spoke off line with @alejo740, It might make sense to use and OKHttp Interceptor here.

Reply:
This ticket is to see if an OKHttp Interceptor can be used...
https://issues.corp.twilio.com/browse/VIDEO-10785